src/main/java/com/google/devtools/build/remote/client/RemoteClient.java

```diff
@@ -173,21 +175,40 @@ private static void printList(List<String> list, int limit) {
     }
   }
 
+  private static Digest toV2(com.google.devtools.remoteexecution.v1test.Digest d) {
+    return Digest.newBuilder().setHash(d.getHash()).setSizeBytes(d.getSizeBytes()).build();
```

Comment:
For binary compatible protos, you could do what I did in RBE and just serialize/deserialize. Up to you, but I think it will be better, because in some cases you are implicitly doing it anyway (e.g. you parse a V2 Command from a V1 proto), so this way will at least be consistent -- and shorter.